src/Network/OWSSignalService.m

```diff
@@ -28,24 +52,105 @@ - (instancetype)init
 
     _censorshipConfiguration = [OWSCensorshipConfiguration new];
 
+    [self observeNotifications];
+
+    [self updateIsCensorshipCircumventionActive];
+
+    OWSSingletonAssert();
+
     return self;
 }
 
-- (BOOL)isCensored
+- (void)observeNotifications
 {
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(registrationStateDidChange:)
+                                                 name:kNSNotificationName_RegistrationStateDidChange
+                                               object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(localNumberDidChange:)
+                                                 name:kNSNotificationName_LocalNumberDidChange
+                                               object:nil];
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)updateHasCensoredPhoneNumber
+{
+    OWSAssert([NSThread isMainThread]);
+
     NSString *localNumber = [TSAccountManager localNumber];
 
     if (localNumber) {
-        return [self.censorshipConfiguration isCensoredPhoneNumber:localNumber];
+        self.hasCensoredPhoneNumber = [self.censorshipConfiguration isCensoredPhoneNumber:localNumber];
     } else {
-        DDLogError(@"no known phone number to check for censorship.");
-        return NO;
+        DDLogError(@"%@ no known phone number to check for censorship.", self.tag);
+        self.hasCensoredPhoneNumber = NO;
+    }
+
+    [self updateIsCensorshipCircumventionActive];
+}
+
+- (BOOL)isCensorshipCircumventionManuallyActivated
+{
+    OWSAssert([NSThread isMainThread]);
+
+    return [[NSUserDefaults.standardUserDefaults
```

Comment:
It didn't seem like a secret so I wasn't going to go to all of that trouble. WDYT?